### ISSUE TITLE
'Cannot access uninitialized non-nullable property' on symfony/var-exporter 6.3.9

### ISSUE THREAD

**Issue by stephanvierkant** [symfony/symfony#52839]
### Symfony version(s) affected

6.3.9, 6.4.0

### Description

As of Symfony/var-exporter 6.3.9 I've got this error message on multiple places in my application:

> Cannot access uninitialized non-nullable property App\\Entity\\User::$notifications by reference

It happens when I get an Organization from Doctrine. 

My entity:
```
#[\ApiPlatform\Metadata\ApiResource(
    // ...
)]
#[\Doctrine\ORM\Mapping\Entity(UserRepository::class)]
class User
{
    #[ManyToOne(Organisation::class, cascade: ['persist'], inversedBy: 'users')]
    #[Gedmo\Versioned]
    private ?Organisation $organisation = null;

    #[OneToMany('user', targetEntity: \App\Entity\User\Notification::class)]
    private Collection $notifications;
    public function __construct()
    {
        $this->notifications = new ArrayCollection();
    }
}

```

### How to reproduce

I'll create a reproducer if necessary, but before I do I'd like to hear some feedback. Maybe it's a simple mistake I've made, or someone has some suggestions how to create a reproducer.

**Comment by nicolas-grekas** [comment]
This might have been solved by #52797. Can you please give it a try? (you might need to clear the cache explicitly before trying again)

**Comment by stephanvierkant** [comment]
It has landed in 6.4.0 hasn't it? If so: nope, that didn't fix it. And yes, I've cleared the cache.

**Comment by nicolas-grekas** [comment]
In 6.4-dev

**Comment by stephanvierkant** [comment]
I'll try tomorrow!

**Comment by stephanvierkant** [comment]
@nicolas-grekas No, it didn't fix it.

**Comment by nicolas-grekas** [comment]
Could you please create a small app I could clone that reproduces the issue?

**Comment by stephanvierkant** [comment]
I'll do! I've already found out that #52762 caused this issue.

**Comment by xabbuh** [comment]
Can you double-check and confirm that neither 6.3.10 nor 6.4.1 did fix this issue?

**Comment by xabbuh** [comment]
@stephanvierkant Did you have the chance to test it again?

**Comment by stephanvierkant** [comment]
Yes, but it didn't help.

(sorry, didn't find the time to create a reproduceer yet)

**Comment by xabbuh** [comment]
Let's close here for now. We can still consider to reopen once we are able to reproduce it.

**Comment by stephanvierkant** [comment]
I have tried to create a reproducer, but unfortunately, I was not successful. In a simple application, it works fine, so apparently there is something specific in my application that I have not been able to locate yet.

I unfortunately cannot provide access to my application, but if someone is willing to help, I can temporarily share the code. Perhaps someone can assist in identifying where the problem might be.

I'm using magic setters/getters in my entities, maybe this can cause the issue? And the relations (user->notifications) are LAZY.

**Comment by stephanvierkant** [comment]
@xabbuh I think I've found it. I'll try to create a reproducer, but meanwhile let me give a simplified example of my use case:

I've got an Item, Property and ItemProperty entity. So I've got a Car (item), colour/size/whatever (property) and a relationship: red car. Using magic method __set I'm calling `$car->colour = 'red'`. That `__set` method looks up property to see if it has to change to colour (`$property->setValue('red')`) or create a `new ItemProperty`.
